test/integration/coscheduling_test.go

```diff
@@ -0,0 +1,286 @@
+		v1.ResourceMemory: *resource.NewQuantity(300, resource.DecimalSI),
+	}
+	node, err := cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName, err)
+	}
+
+	// Create 6 Pods belongs to two podGroup.
+	// pg1 has three pods, and the resources of the cluster can't meet the min of pg1.
+	// pg2 has three pods, but the resources of the cluster can meet the min of pg2.
+	var pods []*v1.Pod
+	pgNames1 := []string{"pod1-1", "pod1-2", "pod1-3"}
+	pgNames2 := []string{"pod2-1", "pod2-2", "pod2-3"}
+	pause := imageutils.GetPauseImageName()
+	for i := 0; i < len(pgNames1); i++ {
+		pod := st.MakePod().Namespace(ns).Name(pgNames1[i]).Container(pause).Obj()
+		pod.Spec.Containers[0].Resources = v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+			},
+		}
+		pod.Labels = map[string]string{
+			coscheduling.PodGroupName:         "pg1",
+			coscheduling.PodGroupMinAvailable: "3",
+		}
+		pods = append(pods, pod)
+
+		pod = st.MakePod().Namespace(ns).Name(pgNames2[i]).Container(pause).Obj()
+		pod.Spec.Containers[0].Resources = v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+			},
+		}
+		pod.Labels = map[string]string{
+			coscheduling.PodGroupName:         "pg2",
+			coscheduling.PodGroupMinAvailable: "3",
+		}
+		pods = append(pods, pod)
+	}
+
+	// Create 6 Pods, We will expect them to be scheduled in a reversed order.
+	t.Logf("Start to create 6 Pods.")
+	for i := range pods {
+		t.Logf("Creating Pod %q", pods[i].Name)
+		_, err := cs.CoreV1().Pods(pods[i].Namespace).Create(context.TODO(), pods[i], metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to create Pod %q: %v", pods[i].Name, err)
+		}
+	}
+
+	// Wait for all Pods are in the scheduling queue.
+	err = wait.Poll(time.Millisecond*200, wait.ForeverTestTimeout, func() (bool, error) {
+		if len(testCtx.Scheduler.SchedulingQueue.PendingPods()) == len(pods) {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	go testCtx.Scheduler.Run(testCtx.Ctx)
+	if err = wait.Poll(100*time.Millisecond, 10*time.Second, podScheduled(cs, ns, pgNames2[0])); err != nil {
+		t.Errorf("Failed to schedule Pod %q: %v", pgNames2[0], err)
+	}
+	if err = wait.Poll(100*time.Millisecond, 2*time.Second, podScheduled(cs, ns, pgNames2[1])); err != nil {
+		t.Errorf("Failed to schedule Pod %q: %v", pgNames2[1], err)
+	}
+	if err = wait.Poll(100*time.Millisecond, 2*time.Second, podScheduled(cs, ns, pgNames2[2])); err != nil {
+		t.Errorf("Failed to schedule Pod %q: %v", pgNames2[2], err)
+	}
```

Comment:
It'd be good to use a single wait to check all Pods are running instead of three checks.